From 012efff130e1d8c6a9672df3c2c616a405acea0d Mon Sep 17 00:00:00 2001
From: Brian Waldon <bcwaldon@gmail.com>
Date: Tue, 21 Aug 2012 13:07:08 -0700
Subject: [PATCH] Update command descriptions

Several commands did not have descriptions or the descriptions
they had were insufficient. This adds mission descriptions
and fattens up those that were too lean.

Change-Id: I091ae70cdae5d3f72f273519d88873cb5392ba3b
---
 glanceclient/v1/legacy_shell.py |    1 +
 glanceclient/v1/shell.py        |    7 ++++++-
 glanceclient/v2/shell.py        |    2 +-
 3 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/glanceclient/v1/legacy_shell.py b/glanceclient/v1/legacy_shell.py
index 030cee4..b6f3903 100644
--- a/glanceclient/v1/legacy_shell.py
+++ b/glanceclient/v1/legacy_shell.py
@@ -249,6 +249,7 @@ def do_delete(gc, args):
 
 @utils.arg('id', metavar='<IMAGE_ID>', help='ID of image to describe.')
 def do_show(gc, args):
+    """DEPRECATED! Use image-show instead."""
     image = gc.images.get(args.id)
     print_image_formatted(gc, image)
     return SUCCESS
diff --git a/glanceclient/v1/shell.py b/glanceclient/v1/shell.py
index e131a08..3646f9f 100644
--- a/glanceclient/v1/shell.py
+++ b/glanceclient/v1/shell.py
@@ -42,7 +42,7 @@ from glanceclient.v1.legacy_shell import *
 @utils.arg('--page-size', metavar='<SIZE>', default=None, type=int,
            help='Number of images to request in each paginated request.')
 def do_image_list(gc, args):
-    """List images."""
+    """List images you can access."""
     filter_keys = ['name', 'status', 'container_format', 'disk_format',
                'size_min', 'size_max']
     filter_items = [(key, getattr(args, key)) for key in filter_keys]
@@ -132,6 +132,7 @@ def do_image_download(gc, args):
            help=("Arbitrary property to associate with image. "
                  "May be used multiple times."))
 def do_image_create(gc, args):
+    """Create a new image."""
     # Filter out None values
     fields = dict(filter(lambda x: x[1] is not None, vars(args).items()))
 
@@ -202,6 +203,7 @@ def do_image_create(gc, args):
                  "not explicitly set in the update request. Otherwise, "
                  "those properties not referenced are preserved."))
 def do_image_update(gc, args):
+    """Update a specific image."""
     # Filter out None values
     fields = dict(filter(lambda x: x[1] is not None, vars(args).items()))
 
@@ -241,6 +243,7 @@ def do_image_delete(gc, args):
 @utils.arg('--tenant-id', metavar='<TENANT_ID>',
            help='Filter results by a tenant ID.')
 def do_member_list(gc, args):
+    """Describe sharing permissions by image or tenant."""
     if args.image_id and args.tenant_id:
         print 'Unable to filter members by both --image-id and --tenant-id.'
         sys.exit(1)
@@ -264,6 +267,7 @@ def do_member_list(gc, args):
 @utils.arg('--can-share', action='store_true', default=False,
            help='Allow the specified tenant to share this image.')
 def do_member_create(gc, args):
+    """Share a specific image with a tenant."""
     gc.image_members.create(args.image_id, args.tenant_id, args.can_share)
 
 
@@ -272,6 +276,7 @@ def do_member_create(gc, args):
 @utils.arg('tenant_id', metavar='<TENANT_ID>',
            help='Tenant to add as member')
 def do_member_delete(gc, args):
+    """Remove a shared image from a tenant."""
     if not options.dry_run:
         gc.image_members.delete(args.image_id, args.tenant_id)
     else:
diff --git a/glanceclient/v2/shell.py b/glanceclient/v2/shell.py
index 63bcd13..4ed2f32 100644
--- a/glanceclient/v2/shell.py
+++ b/glanceclient/v2/shell.py
@@ -20,7 +20,7 @@ from glanceclient import exc
 @utils.arg('--page-size', metavar='<SIZE>', default=None, type=int,
            help='Number of images to request in each paginated request.')
 def do_image_list(gc, args):
-    """List images."""
+    """List images you can access."""
     kwargs = {}
     if args.page_size is not None:
         kwargs['page_size'] = args.page_size
